File: livingdoc-fixture-api/src/main/java/org/livingdoc/fixture/api/converter/exceptions/ValueFormatException.java
package org.livingdoc.fixture.api.converter.exceptions;

public class ValueFormatException extends ConversionException {

    private static final String MESSAGE = "The value '%s' does not match the expected format: %s";

    public ValueFormatException(String value, String format, Throwable cause) {
        super(String.format(MESSAGE, value, format), cause);
    }

}

File: livingdoc-converters/src/main/kotlin/org/livingdoc/converters/exceptions/ValueFormatException.kt
package org.livingdoc.converters.exceptions

import org.livingdoc.api.conversion.ConversionException

/**
 * This Exception is used whenever the given value doesn't match the required format.
 * @param value the value that doesn't match
 * @param format the format that should have been matched
 */
class ValueFormatException(
    value: String,
    format: String,
    cause: Throwable
) : ConversionException("The value '$value' does not match the expected format: $format", cause)
